frontend/test/metabase/scenarios/question/bookmarks.cy.spec.js

```diff
@@ -0,0 +1,44 @@
+import { restore, sidebar } from "__support__/e2e/cypress";
+import { getSidebarSectionTitle as getSectionTitle } from "__support__/e2e/helpers/e2e-collection-helpers";
+
+describe("scenarios > question > view", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+  });
+
+  describe("bookmarks", () => {
+    it("should add then remove bookmark from question page", () => {
+      // Add bookmark
+      toggleBookmark();
+
+      cy.visit("/collection/root");
+
+      sidebar().within(() => {
+        getSectionTitle("Bookmarks");
+        cy.findByText("Orders");
+      });
+
+      // Remove bookmark
+      toggleBookmark();
+
+      cy.intercept("GET", "/api/collection/root/items?**").as(
+        "fetchRootCollectionItems",
+      );
+
+      cy.visit("/collection/root");
+
+      cy.wait("@fetchRootCollectionItems");
+
+      getSectionTitle("Bookmarks").should("not.exist");
+    });
+  });
+});
+
+function toggleBookmark() {
+  cy.visit("/question/1");
+
+  cy.contains("Orders").click();
+
+  cy.icon("bookmark").click();
```

Comment:
Nothing's stopping Cypress from navigating to the root collection before Metabase has the chance to save/register the change (save/delete) bookmark. Chances are extremely slim, but given the amount of weird flakes I'd feel much more confident if we have some guard in place for this.

You can use `cy.intercept("/api/bookmark/card/*);` for this. It will match both `POST` and `DELETE` routes.

Reply:
Thank you very much for the review, @nemanjaglumac